common/main/java/com/couchbase/lite/AbstractReplicatorConfiguration.java

```diff
@@ -133,6 +123,9 @@ private static Map<Collection, CollectionConfiguration> copyConfigs(
     @Nullable
     private Database database;
 
+    // Yuggg: what a kludge.
+    boolean maxAttemptsSet;
```

Comment:
You could mimic the C# Nullable<bool> type I suppose which would at least keep this kludgery contained inside a separate class.